File: 8-kyu-kata/beginner-series-number-1-school-paperwork/python/beginner_series_number_1_school_paperwork_test.py
# 8 kyu - Beginner Series #1 School Paperwork  [ ID: 55f9b48403f6b87a7c0000bd  (beginner-series-number-1-school-paperwork) ]
# URL: https://www.codewars.com/kata/55f9b48403f6b87a7c0000bd
# Category: REFERENCE  |  Tags: FUNDAMENTALS
# ******************************************************************************
import codewars_test as test
from solution import paperwork


@test.describe("Fixed Tests")
def basic_tests():
    @test.it("Basic Test Cases")
    def basic_test_cases():
        test.assert_equals(paperwork(5, 5), 25, "Failed at Paperwork(5,5)")
        test.assert_equals(paperwork(5, -5), 0, "Failed at Paperwork(5,-5)")
        test.assert_equals(paperwork(-5, -5), 0, "Failed at Paperwork(-5,-5)")
        test.assert_equals(paperwork(-5, 5), 0, "Failed at Paperwork(-5,5)")
        test.assert_equals(paperwork(5, 0), 0, "Failed at Paperwork(5,0)")


@test.describe("Random Tests")
def random_tests():

    from random import randint

    sol = lambda m, n: m * n if m > 0 and n > 0 else 0

    for c in range(100):
        i = randint(-100, 100)
        j = randint(-100, 100)

        @test.it(f"Testing for paperwork({i}, {j})")
        def test_case():
            test.assert_equals(paperwork(i, j), sol(i, j))

File: kata-8-kyu/beginner-series-number-1-school-paperwork/javascript/beginnerSeriesNumber1SchoolPaperwork.Test.js
//+ ====================================================================================================================
//+
//+ 8 kyu - Beginner Series #1 School Paperwork  [ ID: 55f9b48403f6b87a7c0000bd ] (beginner-series-number-1-school-paperwork)
//+ URL: https://www.codewars.com/kata/55f9b48403f6b87a7c0000bd
//+ Category: REFERENCE  |  Tags: FUNDAMENTALS
//+
//+ ====================================================================================================================

import { assert } from "vitest"
const { paperwork } = require("./beginnerSeriesNumber1SchoolPaperwork")

describe("Simple Tests", function () {
  it("Testing for fixed tests", () => {
    assert.strictEqual(paperwork(5, 5), 25, "Failed at paperwork(5, 5)")
    assert.strictEqual(paperwork(5, -5), 0, "Failed at paperwork(5, -5)")
    assert.strictEqual(paperwork(-5, -5), 0, "Failed at paperwork(-5, -5)")
    assert.strictEqual(paperwork(-5, 5), 0, "Failed at paperwork(-5, 5)")
    assert.strictEqual(paperwork(5, 0), 0, "Failed at paperwork(5, 0)")
  })
})

describe("Random Tests", function () {
  let paperwork2 = (n, m) => (n * m > 0 && n > 0 ? n * m : 0)

  it("Testing for 100 random tests", () => {
    for (let i = 0; i < 100; i++) {
      let a = ~~(Math.random() * 201) - 100
      let b = ~~(Math.random() * 201) - 100
      assert.strictEqual(paperwork(a, b), paperwork2(a, b), `Testing for n = ${a}, m = ${b}`)
    }
  })
})
